cpp/arcticdb/storage/azure/azure_client_impl.cpp

```diff
@@ -30,18 +61,56 @@ Azure::Core::Context get_context(unsigned int request_timeout){
-            curl_transport_options.CAInfo = conf.ca_cert_path();
-        }
-        if (!conf.ca_cert_dir().empty()) {
-            curl_transport_options.CAPath = conf.ca_cert_dir();
-        }
-        client_options.Transport.Transport = std::make_shared<Azure::Core::Http::CurlTransport>(curl_transport_options);
+Azure::Core::Http::TransportOptions get_client_options(const Config &conf) {
+    Azure::Core::Http::TransportOptions client_options;
+
+#if defined(_WIN32)
+    // On Windows, always use WinHTTP
+    ARCTICDB_RUNTIME_DEBUG(log::storage(), "Using WinHTTP transport");
+    if (conf.ca_cert_path().empty() && conf.ca_cert_dir().empty()) {
+        client_options.Transport.Transport = std::make_shared<Azure::Core::Http::WinHttpTransport>();
+    } else {
+        throw ArcticSpecificException<ErrorCode::E_INVALID_USER_ARGUMENT>(
+            "CA certificate settings are not supported on Windows. "
+            "Please use Windows Certificate Manager (certmgr.msc) to manage certificates:\n"
+            "1. Open Certificate Manager (certmgr.msc)\n"
+            "2. Navigate to 'Trusted Root Certification Authorities'\n"
+            "3. Right-click and select 'All Tasks > Import'\n"
+            "4. Select your certificate file and follow the import wizard\n"
+            "5. Ensure 'Place all certificates in the following store' is selected\n"
+            "6. Complete the import by clicking 'Next' and 'Finish'"
+        );
+    }
+#elif defined(__APPLE__)
+    // On macOS, always use libcurl but ignore CA cert paths
+    ARCTICDB_RUNTIME_DEBUG(log::storage(), "Using libcurl transport");
+    if (conf.ca_cert_path().empty() && conf.ca_cert_dir().empty()) {
+        client_options.Transport.Transport = std::make_shared<Azure::Core::Http::CurlTransport>();
+    } else {
+        throw ArcticSpecificException<ErrorCode::E_INVALID_USER_ARGUMENT>(
+            "CA certificate settings are not supported on macOS. "
+            "Please use Keychain Access to manage certificates:\n"
+            "1. Open Keychain Access (Applications > Utilities > Keychain Access)\n"
+            "2. Select 'System' keychain from the left sidebar\n"
+            "3. Click File > Import Items\n"
+            "4. Select your certificate file\n"
+            "5. Enter your keychain password if prompted\n"
+            "6. The certificate will be added to the System keychain"
+        );
     }
+#else
+    // On Linux, use libcurl with CA cert configuration
+    ARCTICDB_RUNTIME_DEBUG(log::storage(), "Using libcurl transport");
+    Azure::Core::Http::CurlTransportOptions curl_transport_options;
+    if (!conf.ca_cert_path().empty()) {
+        curl_transport_options.CAInfo = conf.ca_cert_path();
+    }
+    if (!conf.ca_cert_dir().empty()) {
+        curl_transport_options.CAPath = conf.ca_cert_dir();
+    }
+    client_options.Transport.Transport = std::make_shared<Azure::Core::Http::CurlTransport>(curl_transport_options);
```

Comment:
This line has not changed. Is a new version of `azure-core-cpp` pulled via vcpkg?